app/helpers/course_summaries_helper.rb

```diff
@@ -0,0 +1,156 @@
+module CourseSummariesHelper
+  include SubmissionsHelper
+
+  # Get JSON data for the table
+  def get_table_json_data
+    all_students = Student.includes(:memberships, groupings: :current_submission_used, grade_entry_students: :grades)
+    all_assignments = Assignment.all
+    all_grade_entry_forms = GradeEntryForm.all
+    all_schemes = MarkingScheme.all
+
+    weights = get_marking_weights_for_all_marking_schemes
+    gef_weights = get_gef_marking_weights_for_all_marking_schemes
+
+    max_marks = Hash[Assignment.all.map do |a|
+                  [a.id, get_max_mark_for_assignment(a.id)]
+                end
+                ]
+    gef_max_marks = Hash[GradeEntryForm.all.map do |gef|
+                      [gef.id, get_max_mark_for_grade_entry_form(gef.id)]
+                    end
```

Comment:
`end` at 20, 20 is not aligned with `gef_max_marks = Hash[GradeEntryForm.all.map do |gef|` at 18, 4